benchmark/Benchmark/TomlData.hs

```diff
@@ -0,0 +1,91 @@
+module Benchmark.TomlData
+       ( test
+       , codecToml
+       , tomlData
+       ) where
+
+import Data.Text (Text)
+import Control.Arrow ((>>>))
+
+import Toml (TomlCodec, pretty, (.=), (<!>))
+import Toml.Edsl (mkToml, table, (=:))
+import Toml.Type (TOML (..), Value (..))
+
+import qualified Data.Text.IO as TIO
+import qualified Toml
+
+
+test :: IO ()
+test = do
+    tomlFile <- TIO.readFile "./benchmark/benchmark.toml"
+    TIO.putStrLn $ case Toml.decode codecToml tomlFile of
+        Left msg   -> Toml.prettyException msg
+        Right hask -> Toml.encode codecToml hask
+    TIO.putStrLn $ case Toml.runTomlCodec codecToml tomlData of
+        Left msg   -> Toml.prettyException msg
+        Right hask -> pretty $ Toml.execTomlCodec codecToml hask
+
+data Test = Test
+    { title  :: Text
+    , atom   :: Double
+    , cash   :: Bool
+    , chars  :: [Text]
+    , digits :: [Double]
+    , bool   :: [Bool]
+    , fruit  :: FruitInside
+    , size   :: SizeInside
+    }
+
+data FruitInside = FruitInside
+    { name :: Text
+    , description :: Text
+    }
+
+insideF :: TomlCodec FruitInside
+insideF = FruitInside <$> (Toml.text "name" .= name) <*> (Toml.text "description" .= description)
+
+newtype SizeInside = SizeInside
+    { unSize :: [Either [Text] [Double]]}
+
+insideS :: TomlCodec SizeInside
+insideS = Toml.dimap unSize SizeInside $ eitherTD "distance"
+
+codecToml :: TomlCodec Test
+codecToml = Test
+    <$> Toml.text "title" .= title
+    <*> Toml.double "atom" .= atom
+    <*> Toml.bool "cash" .= cash
+    <*> Toml.arrayOf Toml._Text "chars" .= chars
+    <*> Toml.arrayOf Toml._Double "digits" .= digits
+    <*> Toml.arrayOf Toml._Bool "bool" .= bool
+    <*> Toml.table insideF "fruit" .= fruit
+    <*> Toml.table insideS "size" .= size
+
+eitherTD :: Toml.Key -> TomlCodec [Either [Text] [Double]]
+eitherTD = Toml.arrayOf (Toml._Left >>> (Toml._Array Toml._Text))
+         <!> Toml.arrayOf (Toml._Right >>> (Toml._Array Toml._Double))
```

Comment:
I would prefer to not mix array of arrays and sum types. Let's have the following field inside `Test` data type instead of `SizeInside`

```haskell
testDimensions :: [[Double]]
```